src/command_controller.rs

```diff
@@ -0,0 +1,329 @@
+//!   scope: Namespaced
+//!   versions:
+//!     - name: v1
+//!       served: true
+//!       storage: true
+//!       schema:
+//!         openAPIV3Schema:
+//!           type: object
+//!           properties:
+//!             spec:
+//!               type: object
+//!               properties:
+//!                 name:
+//!                   type: string
+//!             status:
+//!               nullable: true
+//!               type: object
+//!               properties:
+//!                 startedAt:
+//!                   type: string
+//!                 finishedAt:
+//!                   type: string
+//!                 message:
+//!                   type: string
+//!       subresources:
+//!         status: {}";
+//! }
+//!
+//!
+//! #[tokio::main]
+//! async fn main() -> Result<(),Error> {
+//!    stackable_operator::logging::initialize_logging("FOO_OPERATOR_LOG");
+//!    let client = client::create_client(Some("foo.stackable.tech".to_string())).await?;
+//!
+//!    stackable_operator::crd::ensure_crd_created::<FooCluster>(client.clone()).await?;
+//!    stackable_operator::crd::ensure_crd_created::<Bar>(client.clone()).await?;
+//!
+//!    tokio::join!(
+//!        // create main custom resource controller like ...
+//!        // stackable_foocluster_operator.create_controller(client.clone());
+//!        // create command controller
+//!        stackable_operator::command_controller::create_command_controller::<Bar, FooCluster>(client)
+//!    );
+//!    Ok(())
+//! }
+//! ```
+//!
+use crate::client::Client;
+use crate::controller::{Controller, ControllerStrategy, ReconciliationState};
+use crate::error::{Error, OperatorResult};
+use crate::metadata;
+use crate::reconcile::{ReconcileFunctionAction, ReconcileResult, ReconciliationContext};
+use async_trait::async_trait;
+use json_patch::{AddOperation, PatchOperation};
+use kube::api::{ListParams, Meta};
+use kube::Api;
+use serde::de::DeserializeOwned;
+use std::fmt::Debug;
+use std::future::Future;
+use std::pin::Pin;
+use std::time::Duration;
+use tracing::trace;
+
+/// Trait for all commands to be implemented. We need to retrieve the name of the
+/// main controller custom resource.
+/// The referenced resource has to be in the same namespace as the command itself.
+pub trait Command {
+    /// Retrieve the potential "Owner" name of this custom resource
+    fn get_owner_name(&self) -> String;
+}
+
+struct CommandState<C, O>
+where
+    C: Command + Clone + DeserializeOwned + Meta,
+    O: Clone + DeserializeOwned + Meta,
+{
+    context: ReconciliationContext<C>,
+    owner: Option<O>,
+}
+
+impl<C, O> CommandState<C, O>
+where
+    C: Command + Clone + DeserializeOwned + Meta,
+    O: Clone + DeserializeOwned + Meta,
+{
+    /// Check if our custom resource command already has the owner reference set to the main
+    /// controller custom resource. If so we can stop the reconcile.
+    async fn owner_reference_existing(&mut self) -> ReconcileResult<Error> {
```

Comment:
Can you use controller_ref::get_controller_of here?
Because yours doesn't check for controller = true

That'd get you the controller and then you could check kind & name on that one.